### ISSUE TITLE
How does one set the height of a calendar day row?

### ISSUE THREAD

**Issue by kristophM** [patchthecode/JTAppleCalendar#53]
When I display the calendar in .Vertical mode, it stretches the day cell into a tall rectangle, when I want it to be a square. I can share more details if needed. Thanks!

**Comment by patchthecode** [comment]
The cells stretch to the shape of your calendar to fill it. So if your height is greater than your width (a rectangle) then yours cells will be a rectangle as well. 

Are you saying you have a situation where you want the calendar fill the whole screen, but still have the cell height be the same?
I thought about this when I was creating the calendar, but then i thought: Why would someone want the calendar fill the whole screen but not use the whole screen's height (because of a smaller cell height)?

You if you can let me know your situation you need this i can see what i can do.

**Comment by patchthecode** [comment]
OK after re-reading your question. This seems like an issue only for devs that want the calendar take up the entire screen of the app.

Currently, the logic is this: A month will take up the whole view neatly. This was done so that devs can develop a calendar where only **one month** will be visible on the screen. 

But seems your problem is that, since the month is taking up the entire screen, and your view is taking up the whole iPhone screen, then the dayViews gets stretched. Is this your issue?

**Comment by kristophM** [comment]
Hi there, thanks for the speedy response! Here's my issue: My design for the calendar looks like [this](https://www.dropbox.com/s/wxribpp90e182rw/Screenshot%202016-06-23%2016.39.55.png?dl=0). Notice how there are a lot of cells with equal height and width. 

However, when I implement JTAppleCalendar according to the tutorial, I get [this](https://www.dropbox.com/s/4dj4u41rfnch5ob/Screenshot%202016-06-23%2016.42.01.png?dl=0) (Ignore the bad graphics at the bottom!) Notice that there is a lot of space in between rows. At first I thought it was due to the cell shape (which is was, but I was able to set the .frame property on the cell to equal width and height), but then I realized there's actually a lot of space in between the rows. 

I tried increasing the numberOfRows attribute in the configureCalendar function but it didn't have any effect beyond 6 rows. 

Any advice? Thanks!

**Comment by patchthecode** [comment]
The cells will stretch to the length of your calendar view. I see that your calendar's height is taking up almost the entire length of your iPhone screen. Therefore, your cells stretched to accommodate a height of 6 (vertical) cells per screen.

What you need is a customized height of your cell. One Which i do not provide right now. But if you check sometime over the weekend, I can get something implemented for you. I did not think about vertical users because my project was more Horizontal. 😸

**Comment by kristophM** [comment]
Thanks. Is 6 the max number of rows per screen? I've tried, 7, 8, 20 and nothing happens- it remains at 6. Yes, over the weekend would be great if you could give a customized height!

This calendar tool is amazing!

**Comment by patchthecode** [comment]
Yes, the minimum rows = 1 
and maximum rows = 6

the idea was to have a screen of 7 columns (s,m,t,w,t,f,s) with 6 rows maximum.
Will see what i can do over the weekend.
And Thanks for the compliment.

**Comment by patchthecode** [comment]
The update has been pushed to master branch. Is it possible you can test it before release?

If all is well from your end, version 4.0.4 will now enable you to set the size of your cell using this:
```swift
calendarView.itemSize = anyNumber
```
**Note:**
Keep in mind that making the cells smaller will put more cells in view on the screen. More cells on the screen with all sorts of fancy UI means more calculations. So be efficient with your cell view codes so performance will be nice on older devices. I have tested on an old iPhone 4s and performance is good.

Finally: Feel free to give this control a Star ★ rating if you like it. More upgrades are on the way.

**Comment by patchthecode** [comment]
Very grateful for the tip btw. Thanks you!
Did the update work?

**Comment by patchthecode** [comment]
Version 4.1.0 released.
@kristophM you can change the size of your cells now. using
```swift
calendarView.itemSize = someVal
```

Check out the example app attach with this project for an example

**Comment by kristophM** [comment]
Working on implementing it tomorrow. Sorry for the delay!

On Fri, Jul 1, 2016 at 2:46 PM, JTAppleCalendar <notifications@github.com>
wrote:

> Closed #53 <https://github.com/patchthecode/JTAppleCalendar/issues/53>.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/patchthecode/JTAppleCalendar/issues/53#event-710835717>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ADXXlGPTezy5qH0cW9gmSh7-CZ9FQchGks5qRWB7gaJpZM4I87W6>
> .
>



-- 
Kristoph Matthews
Founder, Boxbee <http://boxbeeinc.com> - We're hiring
<https://boxbee.com/jobs>!
A CNN Top Startup
<http://www.cnn.com/interactive/2014/10/tech/cnn10-startups/index.html?hpt=te_r2>
415.754.0699
